### DISCUSSION TITLE
How to add a version info on Mac app

### DISCUSSION THREAD

**Discussion by nai-kon** [pyinstaller/pyinstaller#7867] [Help] [answered]
I'm building Mac/Windows app using pyinstaller.
On Windows exe, I was able to add version number by using `--version-file` option.
[reference]
https://pyinstaller.org/en/stable/usage.html#cmdoption-version-file

However, I can't find any way to add version on building Mac app.
Is there any way to add version on Mac app?

**Comment by rokm** [comment, marked as answer]
Assuming you are building macOS .app bundle, you can set the .app bundle version in the spec file. 

One way is to create a dictionary that contains corresponding entries for `Info.plist` (e.g., `CFBundleVersion`), and then pass that dictionary to `BUNDLE` via `info_plist` argument. See [here](https://github.com/qutebrowser/qutebrowser/blob/main/misc/qutebrowser.spec) for an example. 

You can also directly pass the version string via the [`version` argument](https://github.com/pyinstaller/pyinstaller/blob/a5c17e4d60dcc281eb052042aa5ac84d062870d4/PyInstaller/building/osx.py#L61) to the `BUNDLE`; e.g.,
```
app = BUNDLE(
    ...
    bundle_identifier='org.mycompany.myapp',
    version='1.2.3',
)
```
This will be mapped to [`CFBundleShortVersionString` in the generated Info.plist](https://github.com/pyinstaller/pyinstaller/blob/a5c17e4d60dcc281eb052042aa5ac84d062870d4/PyInstaller/building/osx.py#L579).

Or, you can manually edit the `Info.plist` in the generated .app bundle.

**Comment by nai-kon** [comment]
Thank you for your answer.
I add version string to the BUNDLE section of spec file and problem was solved!
